language/vm/src/deserializer.rs

```diff
@@ -380,10 +380,13 @@ fn build_script_tables(
         match table.kind {
             TableType::MAIN => {
                 let start: usize = table.offset as usize;
-                let end: usize = start + table.count as usize;
-                let mut cursor = Cursor::new(&binary[start..end]);
-                let main = load_function_def(&mut cursor)?;
-                script.main = main;
+                if let Some(end) = start.checked_add(table.count as usize) {
```

Comment:
the idea here was that check_tables() did the verification that indexes are in bounds so I am not sure the check here is significant, just pointing out as I am not sure how things like this (a function that verifies invariants called at some point) should be dealt with in MIRAI

Reply:
For the next few months, at the very least, MIRAI has no support for tracking invariants. So, in its current state, every function has to require the invariants it relies on as part of its preconditions.

Reply:
In this particular case, the loop adds an additional complication since quantifiers are also not yet supported.

Reply:
Thanks for pointing that out. I'll change that check to be an assumption with an accompanying comment explaining why it holds.